### PR TITLE
CLM update 2

### PR DESCRIPTION
**Pull request by timhoar**
## Description:

I have added a standalone converter to replace the indeterminate values in CLM restart file unused snow layers with the `_FillValue` so that when the native DART programs read the file, the values get replaced with the DART MISSING_R8 value.

### Fixes issue

It relates to, but does not totally satisfy Issue 224.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Documentation changes needed?

- [ X] My change requires a change to the documentation.
  - [X ] I have updated the documentation accordingly.
  
The clm readme.rst will still need to reflect _when_ to use the program in the complete description, 
but the clm_to_dart.rst does exist and shows _how_ to use the program.

### Tests
Please describe any tests you ran to verify your changes.

Brett created a low-resolution global test case in  /glade/scratch/bmraczka/CLM5.0.34/10x15

1.  I dumped several variables with Fortran format into a text file
2.  copied that restart file to a new file
3.  ran clm_to_dart
4.  dumped the same variables with the same format to a different text file
5.  compared the two with 'diffuse'

## Checklist for merging

- [ ] Updated changelog entry
- [x] Documentation updated
- [ ] Version tag 

This does not need a tag or changelog entry to go onto the clm_main branch.

## Testing Datasets

- [ ] Dataset needed for testing available upon request
- [x] Dataset download instructions included
- [ ] No dataset needed

Get the restart file in:
/glade/scratch/bmraczka/CLM5.0.34/10x15

### CODE REVIEW

**Comment by timhoar** [comment]
The CLM model_mod.f90 has large chunks of code that will likely be removed. I have not pruned out unused routines yet. (DART_get_var() for example) Don't bother looking too closely at the model_mod.f90 ... until further notice.

File: observations/obs_converters/SIF/work/SIF.input.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+../data/SIF005clip_201807.nc
+../data/SIF005clip_201808.nc
```
Comment (timhoar) [inline observations/obs_converters/SIF/work/SIF.input.txt:2]: Grrrr .... this is a mistake ... I am tempted to check out the 'main' version of this file, but am unclear what will happen when we go to merge 'clm_main' onto 'main'.  Any ideas?  I believe 'git remove' from my branch will not be the right solution when it comes time to merge 'clm_main' onto 'main'.

File: models/clm/readme.rst [hunk 2]
```diff
@@ -1,40 +1,195 @@
...
 
-.. attention::
-
-   ``CLM`` has a private development branch with some features that are delaying its integration with the rest
-   of the public DART repository.  Until that time, you should consider this documentation as out-of-date.
-   There are several groups that are successfully using recent versions of CLM and recent versions of DART.  
-   If you are interested in using ``CLM`` with more recent versions of DART, contact DAReS staff.
-
+|CLM gridcell breakdown|                              
+Figure from the CLM5.0 Tech Note. The image is a link to the document.
```
Comment (timhoar) [inline models/clm/readme.rst:5]: Please do not review the scripting section of this file.  The scripting updates for the supported release will be covered in another pull request.

File: models/clm/readme.rst [hunk 3]
```diff
@@ -1,40 +1,195 @@
...
+
+Configuring an Experiment
+-------------------------
+
+Our notes on how to set up, configure, build, and run CESM for an assimilation 
+experiment evolved into scripts. These scripts are not intended to be a 'black box'; 
+you will have to read and understand them and modify them to your own purpose. 
+They are heavily commented -- in keeping with their origins as a set of notes. 
+If you would like to offer suggestions on how to improve those notes - please 
+send them to dart@ucar.edu - we'd love to hear them.
```
Comment (timhoar) [inline models/clm/readme.rst:192]: This next section is the section to skip ... I have not changed the table describing the scripts ...

**Review by timhoar — COMMENTED** [review]
Please review ASAP.  I only have 10 days left.
I will also push this branch to the development repo so Xueli, Andy,  and Daniel Hagan can review it.

**Comment by nancycollins** [comment]
> Please review ASAP. I only have 10 days left.
> I will also push this branch to the development repo so Xueli, Andy, and Daniel Hagan can review it.

if i check out this branch (clm_main) does it run, or are there still pieces missing to do an end-to-end test?

**Comment by timhoar** [comment]
> > Please review ASAP. I only have 10 days left.
> > I will also push this branch to the development repo so Xueli, Andy, and Daniel Hagan can review it.
> 
> if i check out this branch (clm_main) does it run, or are there still pieces missing to do an end-to-end test?

The branch to check out is 'clm_update_2'.  The scripting is not yet updated, so - no - you cannot do an end-to-end test.
The 'Testing Datasets' section of the PR has the location of the files I used to test - with one exception.  That file had an unsupported date in it, I had to manually modify the date to be from year 2003 (not '3').  I grabbed the obs_seq.in from the test dataset tar file and modfied the date to match the 2003 date. 

My  /glade/work/thoar/git/DART_public/models/clm/work/TestCLM.csh  script  might help with a testing procedure.  You might have to fill in some blanks ...

**Review by nancycollins — APPROVED** [review]
most of my suggestions are for more accurate comments, so i don't think any of them are mandatory.  i didn't review the model_mod.f90 changes at all because they are too extensive.  i think testing the new version is going to be more promising than trying to do a line-by-line review.  so i think this pull request should move onto the clm_main branch and be run to verify the correctness.

File: models/clm/model_mod.f90 [hunk 4]
```diff
@@ -4644,13 +3107,547 @@ subroutine say(what)
...
+enddo OTHER
+
+end subroutine loop_column_fastest
+
+
 !===================================================================
 ! End of model_mod
 !===================================================================
+
 end module model_mod
```
[the remark this answers is not in the document]
Comment (timhoar) [reply models/clm/model_mod.f90:3652]: ABSOLUTELY ... the file _changes_ should no be reviewed. When I get the scripting finished, it will be possible to test the model_mod in a more rigorous fashion. Without a diverse set of input files, it is hard to test "everything".

File: models/clm/dart_to_clm.f90 [hunk 5]
```diff
@@ -1,112 +1,320 @@
...
-! purpose: interface between DART and the CLM model
-!
-! method: Read DART state vector and overwrite values in a CLM restart file.
+! purpose: Update the CLM restart file with the DART posterior. If need
+!          be, take the posterior SWE (a diagnostic variable) and
+!          repartition it into prognostic snow layers.
 !
+! method: Read DART posterior and replace the valid values in the
+!         CLM restart file. The 'clm_to_dart' program replaced the 
+!         CLM indeterminate values with the DART MISSING_R8. Since
```
[the remark this answers is not in the document]
Comment (timhoar) [reply models/clm/dart_to_clm.f90]: I have added another paragraph to describe why dart_to_clm.f90 is needed and what it is trying to achieve.

**Comment by hkershaw-brown** [comment]
couple of questions on the restart files.
1. SNOW_DEPTH vs SNOWDP

clm_to_dart is expecting SNOW_DEPTH.  The older CLM test files we have in /glade/p/cisl/dares/DART_test_cases/clm/ have SNOWDP.  Is SNOWDP something we have to worry about?

2. The time in `dart_to_clm`.  I tried a new clm (cesm case), and that is also bombing  in static_init_model with 

```
 message: year,mon,day,hour,min,sec    1    3    7    0    0    0 not a valid date
```

so is this the model_mod.f90 that needs updating, or some mismatch between the clm restart file and what DART is expecting?

**Comment by timhoar** [comment]
1) The namelist changes are simply to explore as many permutations of the
variables as possible - within reason. The changes related to  SNOW_DEPTH
vs. SNOWDP ... I don't think we should worry about it. Only the history
files have both variables and the difference between the two is subtle. So
depending on which file the person wants to pull it from, they'll have to
consider the source:

float SNOWDP(time, lat, lon) ;
float SNOW_DEPTH(time, lat, lon) ;
SNOWDP:long_name = "gridcell mean snow height"
SNOW_DEPTH:long_name = "snow height of snow covered area"

In the XY history file (there is no SNOW_DEPTH)
float SNOWDP(time, column) ;
SNOWDP:long_name = "gridcell mean snow height" ;

The restart file has no SNOWDP ...
double SNOW_DEPTH(column) ;
SNOW_DEPTH:long_name = "snow depth" ;

2) I've intentionally decided that the CLM model_mod only supports dates
for which we might have actual observations. The CLM spinup takes centuries
if not millenia of model-time, so I'm not keen to allow someone to
assimilate during the transition from bare ground to some sort of
ecosystem. We've been advising people to spin up a single state and then
replicate that single state and pair each replicant with a unique
atmospheric forcing and perform another (much shorter) spinup until there
is some sort of useful ensemble spread. This is true even for 'perfect
model' experiments.

There are  DART scripts that perform the CLM spinup (RUN_TYPE=startup) and
scripts that continue from that point (RUN_TYPE=hybrid).

For testing purposes, I believe you can set CLM date to be something that
matches the obs_seq.in (which I can't recall at the moment) via:

./xmlchange RUN_STARTDATE=${run_startdate}

**Comment by hkershaw-brown** [comment]
just to make sure I am on the right page here,  I am looking at /glade/p/cisl/dares/DART_test_cases/clm/clm_full/test_rma/clm4_121_hybrid_test.clm2_0003.r.2004-01-05-00000.nc 
which I think is a restart file. 
This has SNOWDP

```
ncdump  /glade/p/cisl/dares/DART_test_cases/clm/clm_full/test_rma/clm4_121_hybrid_test.clm2_0003.r.2004-01-05-00000.nc |  grep SNOW
	double SNOWDP(column) ;
		SNOWDP:long_name = "snow depth" ;
		SNOWDP:units = "m" ;
 SNOWDP = 2.34590526085188, 2.34478478954858, 2.34605796009798, 
```

but no SNOW_DEPTH

```
ncdump -v SNOW_DEPTH /glade/p/cisl/dares/DART_test_cases/clm/clm_full/test_rma/clm4_121_hybrid_test.clm2_0003.r.2004-01-05-00000.nc
ncdump: SNOW_DEPTH: No such variable
```

Is this a CLM4 vs CLM4 difference?

**Comment by nancycollins** [comment]
tim - your comments about snow_depth vs snow_dp, and also the date issue seem incredibly useful because they say "why" not just what to do or run.  are they in our CLM/DART docs?  if not, i'd think they could be cut and pasted as-is.  explaining why you have to use our scripts and what the issues are when you don't seems really helpful.  also any other "why" information would be good to capture.  anyone can fix bugs in the codemand remove trailing blanks (!) but knowing what the code *should* be doing and *why* is something that's only in your head and should be added to the docs (if missing) before you retire.

**Comment by timhoar** [comment]
Yes, this appears to be a CLM4 vs. CLM5 difference.  The
clm4_121_hybrid_test.clm2_0003.r.2004-01-05-00000.nc   files are, for lack
of a better term ... deprecated.
The most current release tag of CLM right now is from Nov 2020 ...
release-cesm2.2.01

The CLM test files in test_build_datasets_preprocess.tar are from a more
relevant CLM release (release-clm5.0.27) circa Aug 2019.

*0[1430]**main **~/<3>clm/work * % ncdump -h
clm_dart.clm2.r.2013-07-02-00000.nc | grep SNOW

double SNOW_DEPTH(column) ;

SNOW_DEPTH:long_name = "snow depth" ;

SNOW_DEPTH:units = "m" ;

SNOW_DEPTH:interpinic_flag = 1 ;

SNOW_DEPTH:interpinic_flag_meanings = "1=nearest neighbor, 2=copy directly,
3=skip, 4=area" ;

SNOW_DEPTH:varnames_on_old_files = "SNOW_DEPTH" ;

SNOW_DEPTH:_FillValue = 1.e+36 ;

SNOW_DEPTH:missing_value = 1.e+36 ;

File: models/clm/readme.rst [hunk 6]
```diff
@@ -1,40 +1,195 @@
...
+~~~~~~~~~~~~~~~~~~~~~~
+
+The *prognostic* variables for snow (i.e. the ones that impact the forecast) 
+are the ones that have layers. The snow observations are typically without 
+explicit depths and are essentially column-integrated quantities like snow 
+water equivalent (SWE - CLM variable *H2OSNO*) or snow depth 
+(CLM variable *SNOW_DEPTH*).  These CLM *diagnostic* variables  
+simplify the forward operator and have been part of the DART state. 
+However, updating the prognostic variables 
+(*T_SOISNO, H2OSOI_LIQ, H2oSOI_ICE, DZSNO, ZSNO, ZISNO*)
```
Comment (braczka) [inline models/clm/readme.rst:156]:
```suggestion
(*T_SOISNO, H2OSOI_LIQ, H2OSOI_ICE, DZSNO, ZSNO, ZISNO*)
```

File: models/clm/readme.rst [hunk 7]
```diff
@@ -1,40 +1,195 @@
...
+
+
+CLM indeterminate values
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+CLM variables are encoded as rectangular arrays in the netCDF files. (Thankfully!)
+However, this means that some variables have space for layers that are unused.
+Anything with snow layers, for example. CLM has the *SNLSNO* varible to indicate
+which snow layers are active. The *unused* layers may not have the *_FillValue*
+value, but can have 'indeteminate' values. The :doc:`clm_to_dart`
```
Comment (braczka) [inline models/clm/readme.rst:65]:
```suggestion
value, but can have 'indeterminate' values. The :doc:`clm_to_dart` 
```

File: models/clm/readme.rst [hunk 8]
```diff
@@ -1,40 +1,195 @@
...
+
+
+Model Interpolate - The Forward Operator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since the subgridscale components of CLM have no explicit location associated 
+with them, the location of every component in the gridcell is the same as the 
+gridcell itself. The DART forward operators fundamentally rely on 
+interpolating the model state to some arbitrary location. At present, the best we
+can do is to create an area-weighted average of all pieces in the gridcell.
```
Comment (braczka) [inline models/clm/readme.rst:81]: Prior few sentences are important, but a bit vague.   Because subgrid components in CLM are not spatially explicit, given an explicit sub-grid location of a biomass observation (for example), we cannot create a forward operator that spatially interpolates to that precise location.  Currently, any observation within the domain of the grid cell is considered collocated (identical location) as the weighted grid cell average of the land surface property of interest.   This is clearly sub-optimal because this introduces representation error between the grid cell weighted average and the observation.    For this reason, for some applications it may be helpful to aggregate (average) the observations to the same spatial resolution as the CLM grid.  Alternatively, a nice project would be ........

File: models/clm/readme.rst [hunk 9]
```diff
@@ -1,40 +1,195 @@
...
+The forward operator would be more accurate and might have a discernable impact
+on the regression relationship (i.e. ensemble covariance) with the variables
+in the DART state vector.
+
+The *model_interpolate* function in DART achieves efficiency by interpolating
+all the ensemble members at the same time. This gives rise to some challenging
+problems when interpolating values for variables with dynamic active layers.
+Some ensemble members may only have 2 active snow layers, some may have 3 ...
+this is an untenable situation when asked for the snow temperature or water 
+content in layer 3, for example. Consequently - *model_interpolate* will fail
```
Comment (braczka) [inline models/clm/readme.rst:94]: ... but for water content, for example, the situation can become more tenable if we interpolate the column total water content, instead of the individual layer content, correct?  Because interpolating the total water content does not necessitate the existence of all layers across all ensembles -- just the existence of at least 1 layer.  Then if we regress upon the column total water storage, we don't have to have knowledge of the individual layers, but we do have to create a repartitioning algorithm.... we come to this later in documentation.

File: models/clm/readme.rst [hunk 10]
```diff
@@ -1,40 +1,195 @@
...
+~~~~~~~~~~~~
+
+Localization is the term used to restrict the portion of the state to the portion
+believed to be related to the observation. Most often, this is a spatial argument
+but it does not need to be restricted to that. In some way, even the selection
+of the CLM variables to include in the DART state is a de-facto localization.
+Since CLM has such a rich description of land unit types: urban columns, glaciers, 
+lakes, etc. it is also possible (and probably desirable) to explicitly declare
+some columns and/or PFTs to be unaffected by the assimilation - i.e., we are
+going to declare that soil moisture observations should not impact urban columns
```
Comment (braczka) [inline models/clm/readme.rst:108]:
```suggestion
some columns and/or PFTs to be unaffected by the assimilation - i.e., we 
declare that soil moisture observations should not impact urban columns
```

File: models/clm/readme.rst [hunk 11]
```diff
@@ -1,40 +1,195 @@
...
+or deep lakes or ... The **get_close_state()** function employs a routine to
```
Comment (braczka) [inline models/clm/readme.rst:109]: Instead of 'we are going' -- which sounds like a work in progress, suggest 'we declare'.   This is the current way we localize based on land cover type.

File: models/clm/readme.rst [hunk 12]
```diff
@@ -1,40 +1,195 @@
...
+However, updating the prognostic variables 
+(*T_SOISNO, H2OSOI_LIQ, H2OSOI_ICE, DZSNO, ZSNO, ZISNO*)
+through their ensemble covariance with the update to *H2OSNO*, for example
+will generally not result in a posterior SWE (calculated from the prognostic 
+variables) that matches the posterior SWE in *H2OSNO*. Consequently, a 
+partitition function should be used to redistribute the posterior SWE into
+the appropriate prognostic variables. **This is under development.**
+Note that we have not attempted to include any of the snow property
+variables (grain radius, carbon content, etc) in the DART state.
+
```
Comment (braczka) [inline models/clm/readme.rst:164]: I think what we mean is that we 'have not attempted to include any *non-water related* snow property variables (grain radius, carbon content, etc) in the DART state.

File: models/clm/readme.rst [hunk 13]
```diff
@@ -1,40 +1,195 @@
...
+snow - maybe we should just use the values from some other member - but when
+does that stop being acceptable? 10 ensemble members? 20? The distributions
+become multimodal, and the logical end result is that you could wind up using
+1 ensemble member to declare the snow for all the remaining members. That seems
+like a bad idea.
+
+The same logic applies to the variables related to plant growth. If the LAI
+observations indicates there should be something growing and nothing has
+sprouted yet ... DART essentially gives up and moves on ... 
+
```
Comment (braczka) [inline models/clm/readme.rst:182]: I think there should be a distinction here, that the lack of a land surface property (i.e. snow, LAI) for a subset of ensemble members is less likely if we are interested in a grid-cell weighted average of this property because it is far more likely that the property is non-zero, and thus a forward operator could be successfully implemented.   However, the implementation of the update step through regression becomes problematic because all ensemble members are required to calculate the covariance to inform the regression.

This distinction was helpful for me ... not sure if normal user needs this level of detail though.

File: models/clm/readme.rst [hunk 14]
```diff
@@ -253,16 +375,55 @@ The following is perfectly legal (for CLM4, at least):
...
+The solution is to run 
+:doc:`../../assimilation_code/programs/fill_inflation_restart/fill_inflation_restart`
+to create an initial inflation file with inflation values of 1.0 (i.e.
+no inflation). ``fill_inflation_restart`` has been specially designed
+to output inflation values for every variable in the DART state. 
+The idea is to copy the *input* inflation file to the *output* inflation
+file name *before each assimilation cycle*. No new values will be written
+for the variables designated 'NO_COPY_BACK', the original values will persist. 
+
```
Comment (braczka) [inline models/clm/readme.rst:412]: This is a different 'fix' than turning off the adaptive inflation for the first assimilation time step, then turning it on for the remainder of the assimilation window?    Not exactly clear how fill_inflation_restart is implemented as it doesn't appear to be a namelist option.

File: models/clm/readme.rst [hunk 15]
```diff
@@ -1017,44 +575,93 @@ filename               purpose
...
 
-N/A
+References
+----------
 
-.. |CLM gridcell breakdown| image:: ../../guide/images/clm_landcover.jpg
+The 
+`CTSM Documentation <https://escomp.github.io/ctsm-docs/versions/master/html/index.html>`__
+is THE reference for CLM.
+
```
Comment (braczka) [inline models/clm/readme.rst:623]: Should also add link for DART references as well here:  https://dart.ucar.edu/publications/

File: models/clm/readme.rst [hunk 16]
```diff
@@ -1017,44 +575,93 @@ filename               purpose
...
+repartitions it into the respective snow layers in a manner that works with both 
+CLM4.5 and CLM5. This may mean modifying the clm_variables list to contain 
+SNOWDP, H2OSOI_LIQ, H2OSOI_ICE, T_SOISNO, and others that may not be in the UPDATE list.
+#. Implement a fast way to get the quantities needed for the calculation of 
+radiative transfer models - needs a whole column of CLM variables, redundant if 
+multiple frequencies are used.
+#. Figure out what to do when one or more of the ensemble members does not have 
+snow/leaves/etc. when the observation indicates there should be. Ditto for removing 
+snow/leaves/etc. when the observation indicates otherwise.
+#. Right now, the soil moisture observation operator is used by the COSMOS code to
```
Comment (braczka) [inline models/clm/readme.rst:608]: 'forward' operator instead of 'observation' operator?   Also not sure what the relevance is to COSMOS code.

File: models/clm/readme.rst [hunk 17]
```diff
@@ -1017,44 +575,93 @@ filename               purpose
...
-#. Figure out what to do when one or more of the ensemble members does not have snow/leaves/etc. when the observation indicates there should be. Ditto for removing snow/leaves/etc. when the observation indicates otherwise.
-#. Right now, the soil moisture observation operator is used by the COSMOS code to calculate the expected neutron intensity counts. This is the right idea, however, the COSMOS forward operator uses m3/m3 and the CLM units are kg/m2 ... I have not checked to see if they are, in fact, identical. This brings up a bigger issue in that the soil moisture observation operator would also be used to calculate whatever a TDT probe or ??? would measure. What units are they in? Can one operator support both?
+#. Implement a robust update_snow() routine that takes the modified SWE and 
+repartitions it into the respective snow layers in a manner that works with both 
+CLM4.5 and CLM5. This may mean modifying the clm_variables list to contain 
+SNOWDP, H2OSOI_LIQ, H2OSOI_ICE, T_SOISNO, and others that may not be in the UPDATE list.
+#. Implement a fast way to get the quantities needed for the calculation of
```
Comment (braczka) [inline models/clm/readme.rst:602]: This 'auto-enumeration' syntax #.  does not render correctly in html unless (I think) at least 1 item of the list receives an explicit number to reference.   For example, the first item should be 1.,  then #. can be used after that.

File: models/clm/clm_to_dart.rst [hunk 18]
```diff
@@ -0,0 +1,198 @@
+PROGRAM ``clm_to_dart``
+=======================
+
+Overview
+--------
+
+``clm_to_dart`` is used to replace indeterminate values in the CLM restart file
+with the *_FillValue* value specified by the variable attribute. This is necessary
```
Comment (braczka) [inline models/clm/clm_to_dart.rst:8]: Is 'indeterminate' the same as 'missing_value'?

File: models/clm/clm_to_dart.rst [hunk 19]
```diff
@@ -0,0 +1,198 @@
...
+-  namelistfile; ``input.nml``
+
+
+Discussion of Indeterminate Values
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To explore the most robust way to replace the indeterminate values with the 
+DART-required *_FillValue*, a low-resolution run (10 deg x 15 deg) was performed 
+using the **f10_f10_musgs** tag on the **release-clm5.0.34** branch.
+The ``clm_to_dart_nml`` verbosity was set to 3 and the run-time output was captured
```
Comment (braczka) [inline models/clm/clm_to_dart.rst:114]:
```suggestion
The ``clm_to_dart_nml`` verbosity was set to 3 and the ``clm_to_dart``  run-time output was captured 
```

File: models/clm/clm_to_dart.rst [hunk 20]
```diff
@@ -0,0 +1,198 @@
...
+The ``clm_to_dart_nml`` verbosity was set to 3 and the run-time output was captured 
+and is summarized below. The output has been pruned to improve clarity and the 
+column headers have been added. Only the output for the *T_SOISNO* variable is 
+shown as it clearly illustrates the situation.
+
+The **frac_sno** variable (snow cover fraction) is used to determine when
+there is a reliable trace amount of snow and the original values in the snow layer
+closest to the ground are left alone. This was determined to be a more useful
+indicator than **H2OSNO**.  
+
```
Comment (braczka) [inline models/clm/clm_to_dart.rst:123]: In other words all values of T_SOISNO that do not have  frac_sno >0 are changed to _FillValue before DART operates on the file.   Then the original indeterminate value is reinserted back into the file in dart_to_clm

File: models/clm/dart_to_clm.rst [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+      /
+
+
+.. container::
+
+
+   ======================= =================== ================================================================= 
+   Item                    Type                Description                                                     
+   ======================= =================== ================================================================= 
+   dart_to_clm_input_file  character(len=256)  Name of the DART posterior (the output of filter)
```
Comment (braczka) [inline models/clm/dart_to_clm.rst:49]:
```suggestion
   dart_to_clm_input_file  character(len=256)  Name of the DART posterior (the output of the filter program)
```

**Review by braczka — APPROVED** [review]

File: models/clm/readme.rst (hunk 11 above)
Comment (timhoar) [reply models/clm/readme.rst:109]: I changed the wording to be less repetetive, actually.

File: models/clm/readme.rst (hunk 12 above)
Comment (timhoar) [reply models/clm/readme.rst:164]: grain radius could be construed as a water-related property - so I am leaving this.

File: models/clm/clm_to_dart.rst (hunk 18 above)
Comment (timhoar) [reply models/clm/clm_to_dart.rst:8]: not really ...   in the CLM files, the 'missing_value' and the '_FillValue' have the same value. What I'm calling indeterminate are values that are explicitly _not_ '_FillValue' - but should be.
